compiler/src/dmd/constfold.d

```diff
@@ -1271,20 +1271,59 @@ UnionExp Slice(Type type, Expression e1, Expression lwr, Expression upr)
         const uinteger_t ilwr = lwr.toInteger();
         const uinteger_t iupr = upr.toInteger();
         if (sliceBoundsCheck(0, es1.elements.length, ilwr, iupr))
-            cantExp(ue);
+        {
+            bool isError = false;
```

Comment:
```suggestion
            bool isErroneous = false;
```

Reply:
Honestly I dont think that it adds any value and makes the code just harder to read. Plus it feels clumpsy. We should stay at a more easy-to-read name like `isError`. Only other name I could think of whould be `hasError`.